editor/src/clj/editor/gui.clj

```diff
@@ -611,7 +614,8 @@
 
   (property layer g/Str
             (default "")
-            (dynamic edit-type (g/fnk [layer-names] (optional-gui-resource-choicebox layer-names)))
+            (dynamic edit-type (g/fnk [layer-names layer->index]
+                                 (optional-gui-resource-choicebox layer-names (partial sort-by layer->index))))
```

Comment:
There is an `ordered-layer-names` output in `LayersNode`, but it would be a new connection I don't want to add here. Taking into account that we already provide `layer-names` and `layer->index` as well as doing sorting, the current implementation is "safe" performance-wise (no new entities or connection introduced, no new operations I do here).